### PR TITLE
reporter: Respect excludes in notice reporter

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Only collect license notes for packages which are not excluded in the
repository configuration.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/789)

### CODE REVIEW

File: model/src/main/kotlin/AnalyzerResult.kt [hunk 1]
```diff
@@ -56,6 +56,30 @@ data class AnalyzerResult(
         errors.any { it.value.isNotEmpty() }
                 || projects.any { it.scopes.any { it.dependencies.any { it.hasErrors() } } }
     }
+
+    /**
+     * A sorted set containing the [Identifier]s of all packages which are not excluded by the repository configuration.
+     */
+    private val includedPackageIds: SortedSet<Identifier> by lazy {
```
Comment (sschuberth) [inline model/src/main/kotlin/AnalyzerResult.kt]: Nit: Could drop the explicit return type.

File: model/src/main/kotlin/AnalyzerResult.kt [hunk 2]
```diff
@@ -56,6 +56,30 @@ data class AnalyzerResult(
...
+        projects.flatMap { project ->
+            if (project.excluded) {
+                emptyList()
+            } else {
+                project.scopes.flatMap { scope ->
+                    if (scope.excluded) {
+                        emptyList<Identifier>()
```
Comment (sschuberth) [inline model/src/main/kotlin/AnalyzerResult.kt]: Maybe use `sortedSetOf()` (which requires no type then) here to be consistent with the return type of `collectDependencyIds` below?

File: model/src/main/kotlin/AnalyzerResult.kt (hunk 1 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/AnalyzerResult.kt]: Done.

File: model/src/main/kotlin/AnalyzerResult.kt (hunk 2 above)
Comment (mnonnenmacher) [reply model/src/main/kotlin/AnalyzerResult.kt]: Done.

**Pull request merged**